### ISSUE TITLE
May be bug

### ISSUE THREAD

**Issue by KimYoungBo** [tensorflow/tfjs#725]
To get help from the community, check out our [Google group](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs).

#### TensorFlow.js version
0.12.7
#### Browser version
69.0.3497.100
#### Code to reproduce the bug / link to feature request
function train(){
  const varWeight = tf.variable(tf.randomNormal([1], 0, 1, 'float32', 100), true, 'weight');
  for (let i = 0; i < 100; i++){
    optimizer.minimize(() => {
      // disposed
      **tf.dispose(varWeight);**
      return loss(predict());
    });
  };
};

// tf-core 5678 line
function variableGrads(f, varList) {
  if (varList == null) {
    varList = [];
    for (var varName in ENV.engine.registeredVariables) {
      // insert here, reference tf-core 955 line
      _if (this.isDisposed) {
        return;
      }_
      varList.push(ENV.engine.registeredVariables[varName]);
    }
  }
}

// tf-core 955 line
Tensor.prototype.dispose = function () {
  _if (this.isDisposed) {
      return;
  }_
  trackerFn().disposeTensor(this);
  this.isDisposedInternal = true;
};

**Comment by dsmilkov** [comment]
What's the exact issue here? You shouldn't dispose that variable since it's needed to compute the loss.

**Comment by nsthorat** [comment]
Going to close this out for inactivity (we have pretty extensive unit test checking memory allocation here). Please re-open if you think that's a mistake.
